crates/tako/src/tests/utils/resources.rs

```diff
@@ -0,0 +1,57 @@
+use crate::common::resources::{
+    CpuRequest, GenericResourceAmount, GenericResourceId, GenericResourceRequest, NumOfCpus,
+    ResourceRequest,
+};
+use derive_builder::Builder;
+use std::time::Duration;
+
+pub use ResourceRequestConfigBuilder as ResBuilder;
+
+#[derive(Builder, Default, Clone)]
+#[builder(pattern = "owned", derive(Clone))]
+pub struct ResourceRequestConfig {
+    #[builder(default)]
+    cpus: CpuRequest,
+    #[builder(default)]
+    generic: Vec<GenericResourceRequest>,
+    #[builder(default)]
+    min_time: Duration,
+}
+
+impl ResourceRequestConfigBuilder {
+    pub fn add_generic<Id: Into<GenericResourceId>>(
+        mut self,
+        id: Id,
+        amount: GenericResourceAmount,
+    ) -> Self {
+        self.generic
+            .get_or_insert_with(|| vec![])
+            .push(GenericResourceRequest {
+                resource: id.into(),
+                amount,
+            });
+        self
+    }
+
+    pub fn finish(self) -> ResourceRequest {
```

Comment:
Why is it called `finish` and not `build`?

Reply:
There is already a `build` method on the builder, but it returns a `Result`, which is a bit annoying in tests. Actually I see now that the `build` method could be [renamed](https://docs.rs/derive_builder/0.10.2/derive_builder/#build-method-customization), so we could rename the default one and then we can rename this to `build`.

Usually you would have a domain struct `X` and a corresponding builder `XBuilder`. But since we use `derive_builder` and it is only used for tests, we have a domain struct `X`, then a config `XConfig` that mirrors `X` and describes the builder, and then the actual builder `XConfigBuilder`. This method serves to make the additional layer of indirection smaller.

Reply:
Thanks, I understand now.